### PR TITLE
✨ Add metadata for top-level and nested variables & deprecate previous metadata field

### LINKED ISSUE #10776: Variable metadata for top-level & nested variables
Context: we previously introduced top-level metadata for variables in: https://github.com/kubernetes-sigs/cluster-api/issues/10023

Goal of this issue is to also allow adding metadata to nested variables.

The use case is that we want to add metadata for a UI generated from variable schemas not only to top-level variables but also to nested variables (otherwise we have to add all the metadata at the top-level which would be very messay)

I would propose
* To add a `XMetadata` field to `JSONSchemaProps` (of type `VariableSchemaMetadata` (which is a copy of `ClusterClassVariableMetadata`)
* To deprecate `ClusterClassVariable.Metadata`
  * Usually we only add a field below ClusterClassVariable if the same cannot be expressed within the schema. This is now possible

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
I went through both previous PRs and implemented the same just for top-level and nested:
* https://github.com/kubernetes-sigs/cluster-api/pull/10308
* https://github.com/kubernetes-sigs/cluster-api/pull/10337

(please double-check on review that I didn't forget anything)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #10776

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold

(if possible let's merge it after the CEL PR, I want to avoid further rebases on the CEL PR If possible)

**Review by chrischdi — COMMENTED** [review]
One thing but also fine as follow-up

/lgtm

File: api/v1beta1/clusterclass_types.go
Comment (sbueringer) [reply api/v1beta1/clusterclass_types.go:400]: Yes but let's add it to the issue

**Comment by fabriziopandini** [comment]
By adding xMetadata to JSONSchemaProps we got everything included in both old PRs "for free".
Nice to have all the additional unit tests.

/lgtm

**Comment by chrischdi** [comment]
/lgtm

(keeping the approve off until we merged CEL :-) feel free to ping afterwards)

**Comment by sbueringer** [comment]
Yup all good, have to rebase and adjust tests anyway

**Comment by sbueringer** [comment]
/hold cancel

/assign @fabriziopandini @chrischdi 

Ready now

**Review by Danil-Grigorev — APPROVED** [review]
/lgtm

**Review by chrischdi — COMMENTED** [review]
/approve

**Pull request merged**
